lib/finch.ex

```diff
@@ -93,6 +123,18 @@ defmodule Finch do
     Supervisor.init(children, strategy: :one_for_all)
   end
 
+  @doc """
+  Sends an HTTP request and returns the response.
+
+  ## Options:
+    * `:pool_timeout` - (`pos_integer()` | `:infinity`) This timeout is applied when we check out
+    a connection from the pool.
```

Comment:
the built-in [`timeout/0`](https://hexdocs.pm/elixir/typespecs.html#built-in-types) is `non_neg_integer() | :infinity`, should it be like that too? Or, should we just skip it then?

I think we should also show default value.

```suggestion

    * `:pool_timeout` - This timeout is applied when we check out a connection from the pool.
      Default value is `5_000`.
```